.github/workflows/testing.yml

```
@@ -72,6 +74,12 @@ jobs:
           # If we did not exit earlier, raise an error here
           exit 1
           ls
+      - name: Upload Results
+        if: failure()
+        uses: actions/upload-artifact@v3
+        with:
+          name: ${{github.job}}-${{github.run_number}}
+          path: ${{env.MESHPY_TESTING_DIR}}
```

Sorry to be nitpicking, but can't we just do it like this:
```suggestion
      - name: Upload Results
        if: failure()
        uses: actions/upload-artifact@v3
        with:
          name: ${{github.job}}-${{github.run_number}}
          path: ${{github.workspace}}/tests/testing-tmp/
```
Then we don't have to introduce the new variable `MESHPY_TESTING_DIR` which is just used for GitHub testing.